utest/api/test_run_and_rebot.py

```diff
@@ -103,6 +105,19 @@ def test_listener_gets_notification_about_log_report_and_output(self):
                                      ('[log {}]'.format(LOG_PATH), 1),
                                      ('[listener close]', 1)])
 
+    def test_pass_listener_as_instance(self):
+        assert_equals(run(self.data, outputdir=TEMP, report='none', listener=Listener(1)), 1)
+        self._assert_outputs([("[from listener 1]", 1)])
```

Comment:
Good tests! I'd like to enhance this one so that the listener would store some information into itself when it's methods are called and that changed state could be verified after execution. Perhaps the `Listener._log` could be changed to:

```python
    def _log(self, message):
        sys.__stdout__.write("[{}]\n".format(message))
        self.messages.append(message)
```

and that verified later with `assert_equals(listener.message, [...]`.